woocommerce/woocommerce-ios
# REST API: Migrate shipping label endpoints

Closes: #8668 


## Description

This PR enables REST API for all shipping label endpoints and updates the related mappers to parse contents without the data envelope.

My apologies for the size of this PR - most of the changes are for the mock responses.

## Testing instructions

Pre-requisite: Follow the testing guide for setting up shipping label on your self-hosted site: p91TBi-3xD-pp91TBi-3xD2-p2. Then create an order on your store and handle the payment to make it eligible for creating shipping labels.
- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- On the prologue screen, select Enter your site address and proceed with the address of your test store.
- Log in with your site credentials. After the login succeeds, you should be navigated to the home screen.
- Switch to the Orders tab and select the created order. Notice that the button Create shipping label is visible on the order detail screen.
- Tap the Create shipping label button to start the creation. Follow the steps to purchase a label. Everything should work properly (the package selection screen will not display the purchased product since the product endpoints haven't been migrated yet).
- After purchasing a label, navigate back to the Order details screen. The purchased label should be listed on the screen, and you should be able to print it or request a refund.

## Screenshots


https://user-images.githubusercontent.com/5533851/213105965-fa3d02f4-5aca-4494-8989-60d4f39de73c.mp4



---
